m4assembly:

    joint_state_controller:
     type: joint_state_controller/JointStateController
     publish_rate: 30
        
    wheel_velocity_controller:
     type: diff_drive_controller/DiffDriveController
     left_wheel: ['front_left_wheel_joint', 'rear_left_wheel_joint']
     right_wheel: ['front_right_wheel_joint', 'rear_right_wheel_joint']
     publish_rate: 30
     pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
     twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
     cmd_vel_timeout: 0.25
     velocity_rolling_window_size: 2
     # Base frame_id
     base_frame_id: base_link
     enable_odom_tf: false
     wheel_separation: 0.2574
     wheel_radius: 0.127
     # Wheel separation and radius multipliers
     wheel_separation_multiplier: 1.0 # default: 1.0
     wheel_radius_multiplier    : 1.0 # default: 1.0
    
    linear:
     x:
      has_velocity_limits    : true
      max_velocity           : 1.0   # m/s
      has_acceleration_limits: true
      max_acceleration       : 3.0   # m/s^2
    angular:
     z:
      has_velocity_limits    : true
      max_velocity           : 2.0   # rad/s
      has_acceleration_limits: true
      max_acceleration       : 6.0   # rad/s^2
  
    rear_left_hip_position_controller:
     type: effort_controllers/JointPositionController
     joint: rear_left_hip_joint
     publish_rate: 100
     pid: {p: 80, i: 7, d: 1}

    rear_right_hip_position_controller:
     type: effort_controllers/JointPositionController
     joint: rear_right_hip_joint
     publish_rate: 50
     pid: {p: 80, i: 7, d: 1}
    front_left_hip_position_controller:
     type: effort_controllers/JointPositionController
     joint: front_left_hip_joint
     publish_rate: 50
     pid: {p: 80, i: 7, d: 1}

    front_right_hip_position_controller:
     type: effort_controllers/JointPositionController
     joint: front_right_hip_joint
     publish_rate: 50
     pid: {p: 80, i: 7, d: 1}
    
    

